File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/overrideFromMultipleBaseClasses/twoInterfacesAsSuperClass/Base1.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.overrideFromMultipleBaseClasses.twoInterfacesAsSuperClass;

/**
 * Base 1, interface with default method, First base class.
 *
 * @author Ranjith Krishnamurthy
 */
public interface Base1 {
    default public void print1() {
        System.out.println("Base1 --> print1");
    }

    default public void print2() {
        System.out.println("Base1 --> print2");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/overrideFromMultipleBaseClasses/twoInterfacesAsSuperClass/Base1.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.overrideFromMultipleBaseClasses.twoInterfacesAsSuperClass

/**
 * Base 1, interface with default method, First base class.
 *
 * @author Ranjith Krishnamurthy
 */
interface Base1 {
    fun print1() {
        println("Base1 --> print1")
    }

    fun print2() {
        println("Base1 --> print2")
    }
}